files/token_utils.py

```
@@ -0,0 +1,241 @@
+def check_rate_limit(ip: str, friendly_token: str) -> bool:
+    """Check if the IP is rate-limited for this media.
+
+    Returns True if the request is ALLOWED, False if BLOCKED.
+    """
+    key = RATE_LIMIT_KEY_TEMPLATE.format(ip=ip, friendly_token=friendly_token)
+
+    try:
+        redis = _get_redis()
+        attempts = redis.get(key)
+        return not (attempts is not None and int(attempts) >= _get_brute_force_max_attempts())
+    except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)